core/vtk/ttkIntegralLines/ttkIntegralLines.cpp

```diff
@@ -40,12 +43,18 @@ int ttkIntegralLines::FillOutputPortInformation(int port,
   return 1;
 }
 
+template <typename triangulationType>
 int ttkIntegralLines::getTrajectories(
   vtkDataSet *input,
-  ttk::Triangulation *triangulation,
-  std::vector<std::vector<ttk::SimplexId>> &trajectories,
+  const triangulationType *triangulation,
+  const std::vector<
+    ttk::ArrayLinkedList<ttk::intgl::IntegralLine, INTEGRAL_LINE_TABULAR_SIZE>>
+    &integralLines,
+#ifdef TTK_ENABLE_MPI
+  const std::vector<ttk::SimplexId> &globalVertexId,
```

Comment:
quick question: why no moving the `std::vector<ttk::SimplexId> &globalVertexId` and the `std::vector<ttk::SimplexId> &globalCellId` to the `ttk::intgl::IntegralLine`?

Reply:
The `ttk::intgl::IntegralLine` is used all along the computation of the algorithm, while the `globalVertexId` and `globalCellId` vectors are only used afterwards during the computation of the global ids. It wouldn't be efficiently possible to pre-compute the global ids during the computation of the integral lines, therefore they don't need to go together.

Reply:
alright, thanks for the clarification.